community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/DeleteAcceptanceTest.scala

```diff
@@ -272,4 +277,48 @@ class DeleteAcceptanceTest extends ExecutionEngineFunSuite with QueryStatisticsT
     assertStats(result, nodesDeleted = 2, relationshipsDeleted = 2)
   }
 
+  //see https://github.com/neo4j/neo4j/issues/7407
+  test("should plan within reasonable time") {
+    updateWithBothPlannersAndCompatibilityMode("CREATE (n1: `node1` {oid: '1'})")
```

Comment:
this should probably be a single query, and not be run through the `updateWithBothPlannersAndCompatibilityMode` thingie